File: benchmark/c/cpw/number_non_negative_integral_solutions_b_c_n/number_non_negative_integral_solutions_b_c_n.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int number_non_negative_integral_solutions_b_c_n ( int n ) ;
int number_non_negative_integral_solutions_b_c_n ( int n ) {
  int result = 0;
  for ( int i = 0;
  i <= n;
  i ++ ) for ( int j = 0;
  j <= n - i;
  j ++ ) for ( int k = 0;
  k <= ( n - i - j );
  k ++ ) if ( i + j + k == n ) result ++;
  return result;
}

File: translation_result/else_bms/fail/c/cpw/number_non_negative_integral_solutions_b_c_n/gemini/BaseRepair/number_non_negative_integral_solutions_b_c_n.rs

fn number_non_negative_integral_solutions_b_c_n(n: i32) -> i32 {
    let mut result = 0;
    for i in 0..=n {
        for j in 0..=(n - i) {
            for k in 0..=(n - i - j) {
                if i + j + k == n {
                    result += 1;
                }
            }
        }
    }
    result
}
